examples/sha256/main.rs

```diff
@@ -1,162 +1,162 @@
-    pub fn update(
-        &mut self,
-        mut layouter: impl Layouter<Sha256Chip>,
-        mut data: &[Sha256Chip::BlockWord],
-    ) -> Result<(), Error> {
-        self.length += data.len() * 32;
-
-        // Fill the current block, if possible.
-        let remaining = BLOCK_SIZE - self.cur_block.len();
-        let (l, r) = data.split_at(min(remaining, data.len()));
-        self.cur_block.extend_from_slice(l);
-        data = r;
-
-        // If we still don't have a full block, we are done.
-        if self.cur_block.len() < BLOCK_SIZE {
-            return Ok(());
-        }
-
-        // Process the now-full current block.
-        self.state = Sha256Chip::compress(
-            &mut layouter,
-            &self.state,
-            self.cur_block[..]
-                .try_into()
-                .expect("cur_block.len() == BLOCK_SIZE"),
-        )?;
-        self.cur_block.clear();
-
-        // Process any additional full blocks.
-        let mut chunks_iter = data.chunks_exact(BLOCK_SIZE);
-        for chunk in &mut chunks_iter {
-            self.state = Sha256Chip::initialization(&mut layouter, &self.state)?;
-            self.state = Sha256Chip::compress(
-                &mut layouter,
-                &self.state,
-                chunk.try_into().expect("chunk.len() == BLOCK_SIZE"),
-            )?;
-        }
-
-        // Cache the remaining partial block, if any.
-        let rem = chunks_iter.remainder();
-        self.cur_block.extend_from_slice(rem);
-
-        Ok(())
-    }
-
-    /// Retrieve result and consume hasher instance.
-    pub fn finalize(
-        mut self,
-        mut layouter: impl Layouter<Sha256Chip>,
-    ) -> Result<Sha256Digest<Sha256Chip::BlockWord>, Error> {
-        // Pad the remaining block
-        if !self.cur_block.is_empty() {
-            let padding = vec![Sha256Chip::zero(); BLOCK_SIZE - self.cur_block.len()];
-            self.cur_block.extend_from_slice(&padding);
-            self.state = Sha256Chip::initialization(&mut layouter, &self.state)?;
-            self.state = Sha256Chip::compress(
-                &mut layouter,
-                &self.state,
-                self.cur_block[..]
-                    .try_into()
-                    .expect("cur_block.len() == BLOCK_SIZE"),
-            )?;
-        }
-        Sha256Chip::digest(&mut layouter, &self.state).map(Sha256Digest)
-    }
-
-    /// Convenience function to compute hash of the data. It will handle hasher creation,
-    /// data feeding and finalization.
-    pub fn digest(
-        mut layouter: impl Layouter<Sha256Chip>,
-        data: &[Sha256Chip::BlockWord],
-    ) -> Result<Sha256Digest<Sha256Chip::BlockWord>, Error> {
-        let mut hasher = Self::new(layouter.namespace(|| "init"))?;
-        hasher.update(layouter.namespace(|| "update"), data)?;
-        hasher.finalize(layouter.namespace(|| "finalize"))
-    }
-}
+// //! Gadget and chips for the [SHA-256] hash function.
```

Comment:
Use a block comment (`/* ... */`) so that when #256 fixes the example, we can see the actual changes in the diff.